internal/campaigns/executor.go

```diff
@@ -147,8 +147,6 @@ func (x *executor) do(ctx context.Context, task *Task) (err error) {
 		} else if result != nil {
 			status.Cached = true
 			status.ChangesetSpec = result
-			status.FinishedAt = time.Now()
-			x.updateTaskStatus(task, status)
```

Comment:
Was there a reason for removing these (and the `updateTaskStatus` call below)?

Reply:
They're all handled in the `defer` up and I thought it was confusing that at some points in the function we duplicate the `defer` and at others we don't.